scripts/mk_all.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Makes a file all.lean in all subfolders of src/ importing all files in that folder, including subfolders.
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+for d in $(find $DIR/../src/ -maxdepth 99 -type d)
+do
+  cd "$d"
+  find . -maxdepth 1 -mindepth 1 -name 'all.lean' -prune -o -name '*.lean' -print -o -type d -print |
+  sed 's/$/\.all/; s/\.lean\.all//; s/^\.\//       \./; 1s/^      /import/' > all.lean
```

Comment:
What is this `sed`-script doing? It first seems to add `.all` to every line, then removes `.lean.all` again...

Reply:
It is supposed to transform something like
```
./foo.lean
./bar.lean
./foodir
./bardir
```
to
```
import .foo
       .bar
       .foodir.all
       .bardir.all
```
To do this, I first add `.all` to the end of every line, then remove all occurrences of `.lean.all`, then replace `./` by `.` with some spaces and then replace the spaces by `import` on the first line.